### PR TITLE
NO-ISSUE Check that all files are generated in CI

### PR DESCRIPTION
**Pull request by carbonin**

### CODE REVIEW

**Comment by carbonin** [comment]
It seems like we don't use the Jenkinsfile from the PR when we run tests.
@YuviGold how should I go about this so I know the changes will work?
I was assuming this would fail until https://github.com/openshift/assisted-service/pull/708 was merged, but it looks like it won't because we're not running CI using the changes I made.

**Comment by YuviGold** [comment]
That's right. An untrusted can't run Jenkinsfile changes for security reasons.
Let's include it as part of `ci-lint` Makefile target

/hold

**Comment by carbonin** [comment]
I'm not sure which directories to include in the check. It seems to me that we'd get a lot of false failures in mock files.
After running this locally I'm getting a bunch of diffs like this:
```patch
diff --git a/pkg/s3wrapper/mock_s3iface.go b/pkg/s3wrapper/mock_s3iface.go
index 6506a60c..3c374b21 100644
--- a/pkg/s3wrapper/mock_s3iface.go
+++ b/pkg/s3wrapper/mock_s3iface.go
@@ -6,11 +6,10 @@ package s3wrapper
 
 import (
        context "context"
-       reflect "reflect"
-
        request "github.com/aws/aws-sdk-go/aws/request"
        s3 "github.com/aws/aws-sdk-go/service/s3"
        gomock "github.com/golang/mock/gomock"
+       reflect "reflect"
 )
 
 // MockS3API is a mock of S3API interface
```

**Comment by carbonin** [comment]
Looks like `mockgen` isn't available on the Jenkins nodes. @YuviGold

**Comment by YuviGold** [comment]
> I'm not sure which directories to include in the check. It seems to me that we'd get a lot of false failures in mock files.
> After running this locally I'm getting a bunch of diffs like this:
> 
> ```diff
> diff --git a/pkg/s3wrapper/mock_s3iface.go b/pkg/s3wrapper/mock_s3iface.go
> index 6506a60c..3c374b21 100644
> --- a/pkg/s3wrapper/mock_s3iface.go
> +++ b/pkg/s3wrapper/mock_s3iface.go
> @@ -6,11 +6,10 @@ package s3wrapper
>  
>  import (
>         context "context"
> -       reflect "reflect"
> -
>         request "github.com/aws/aws-sdk-go/aws/request"
>         s3 "github.com/aws/aws-sdk-go/service/s3"
>         gomock "github.com/golang/mock/gomock"
> +       reflect "reflect"
>  )
>  
>  // MockS3API is a mock of S3API interface
> ```

This sounds like a real problem. I don't know why but `generate` truly changes these files all the time.
@filanov @eranco74 Do you know why? How it can be stopped in order to verify when we generate?

**Comment by YuviGold** [comment]
> Looks like `mockgen` isn't available on the Jenkins nodes. @YuviGold

Yes. It shouldn't. We should move `ci-lint` to be used inside of the build container.
`skipper make ci-lint` by Jenkins

**Comment by filanov** [comment]
> > I'm not sure which directories to include in the check. It seems to me that we'd get a lot of false failures in mock files.
> > After running this locally I'm getting a bunch of diffs like this:
> > ```diff
> > diff --git a/pkg/s3wrapper/mock_s3iface.go b/pkg/s3wrapper/mock_s3iface.go
> > index 6506a60c..3c374b21 100644
> > --- a/pkg/s3wrapper/mock_s3iface.go
> > +++ b/pkg/s3wrapper/mock_s3iface.go
> > @@ -6,11 +6,10 @@ package s3wrapper
> >  
> >  import (
> >         context "context"
> > -       reflect "reflect"
> > -
> >         request "github.com/aws/aws-sdk-go/aws/request"
> >         s3 "github.com/aws/aws-sdk-go/service/s3"
> >         gomock "github.com/golang/mock/gomock"
> > +       reflect "reflect"
> >  )
> >  
> >  // MockS3API is a mock of S3API interface
> > ```
> 
> This sounds like a real problem. I don't know why but `generate` truly changes these files all the time.
> @filanov @eranco74 Do you know why? How it can be stopped in order to verify when we generate?

We don't control the way the generator is working but try to add `make format` after the generation and it should fix this issue

File: Makefile [hunk 1]
```diff
@@ -110,6 +111,9 @@ generate-python-client: $(BUILD_FOLDER)
 generate-keys: $(BUILD_FOLDER)
 	cd tools && go run auth_keys_generator.go -keys-dir=$(BUILD_FOLDER)
 
+verify-generated-files: generate generate-from-swagger
+	${ROOT_DIR}/tools/verify-generated-files.sh
```
Comment (filanov) [inline Makefile:115]: Add `make format` before the validation it will fix import and lint issues

**Comment by carbonin** [comment]
> Yes. It shouldn't. We should move ci-lint to be used inside of the build container.
> skipper make ci-lint by Jenkins

So now we're back to having to edit the Jenkins file to get this PR to pass. Is there some good way to do that @YuviGold ?

**Comment by YuviGold** [comment]
> > Yes. It shouldn't. We should move ci-lint to be used inside of the build container.
> > skipper make ci-lint by Jenkins
> 
> So now we're back to having to edit the Jenkins file to get this PR to pass. Is there some good way to do that @YuviGold ?

Just in order to give Jenkins more capabilities. That's fine. :)

File: Makefile (hunk 1 above)
Comment (YuviGold) [reply Makefile:115]: Adding `make format` would be cheating the CI. We run this verification in order to verify if the committed changes are legitimate.

Comment (filanov) [reply Makefile:115]: The generated code is not formatted according to `goimports` so adding it the as part of generate target is fine.

Comment (filanov) [reply Makefile:115]: i wrote the comment in the wrong place, it should be part of make generate.

Comment (YuviGold) [reply Makefile:115]: Alright, I see. You're saying we push code without it being formatted.
SGTM

Comment (carbonin) [reply Makefile:115]: I'm not sure what's going on ... I added this:
```patch
diff --git a/Makefile b/Makefile
index 6dc95683..a7ade2f5 100644
--- a/Makefile
+++ b/Makefile
@@ -85,6 +85,7 @@ format:
 
 generate:
        go generate $(shell go list ./... | grep -v 'assisted-service/models\|assisted-service/client\|assisted-service/restapi')
+       $(MAKE) format
 
 generate-from-swagger: generate-go-client generate-go-server
 
```

But now `skipper make generate` fails on `goimports`:
```
goimports -w -l cmd/ internal/ subsystem/ pkg/ assisted-iso-create/
internal/cluster/mock_cluster_api.go
internal/connectivity/mock_connectivity_validator.go
internal/events/mock_event.go
internal/hardware/mock_validator.go
internal/host/mock_host_api.go
internal/host/mock_instruction_api.go
internal/metrics/mock_netricsManager_api.go
internal/network/mock_ntp_utils.go
pkg/generator/mock_install_config.go
pkg/k8sclient/mock_k8sclient.go
pkg/leader/mock_leader_elector.go
pkg/ocm/mock_authorization.go
pkg/ocm/mock_pullsecret_auth.go
pkg/s3wrapper/mock_s3iface.go
pkg/s3wrapper/mock_s3wrapper.go
make[1]: *** [Makefile:79: format] Error 1
make[1]: Leaving directory '/home/ncarboni/Source/assisted-service'
make: *** [Makefile:88: generate] Error 2
```

But if I run `skipper make format`, it succeeds but also doesn't clean up any of the imports issues:
```
$ skipper make format
[skipper] No build container tag was provided
...
Successfully built e7f24fddf102
Successfully tagged assisted-service-build:latest
goimports -w -l cmd/ internal/ subsystem/ pkg/ assisted-iso-create/
gofmt -w -l cmd/ internal/ subsystem/ pkg/ assisted-iso-create/
$ git status -s
 M Makefile
 M internal/cluster/mock_cluster_api.go
 M internal/connectivity/mock_connectivity_validator.go
 M internal/events/mock_event.go
 M internal/hardware/mock_validator.go
 M internal/host/mock_host_api.go
 M internal/host/mock_instruction_api.go
 M internal/metrics/mock_netricsManager_api.go
 M internal/network/mock_ntp_utils.go
 M pkg/generator/mock_install_config.go
 M pkg/k8sclient/mock_k8sclient.go
 M pkg/leader/mock_leader_elector.go
 M pkg/ocm/mock_authorization.go
 M pkg/ocm/mock_pullsecret_auth.go
 M pkg/s3wrapper/mock_s3iface.go
 M pkg/s3wrapper/mock_s3wrapper.go
```
:confused:

Comment (YuviGold) [reply Makefile:115]: IMO we don't really need to upload the mock files, we can generate them each time we run the testing.

Comment (carbonin) [reply Makefile:115]: What do you mean by "upload", commit?
That would be quite a large change for just this ...

Comment (filanov) [reply Makefile:115]: Strange, i just added this code:
```
 generate-from-swagger: generate-go-client generate-go-server
+       make format
```
and everything worked

Comment (carbonin) [reply Makefile:115]: Yeah, that's a different generate command. The one that builds the mocks and has the import ordering issue is `generate`, not `generate-from-swagger`

Comment (filanov) [reply Makefile:115]: Ohh we were talking about generate from swagger no? i put the command after generate from swagger.

Comment (carbonin) [reply Makefile:115]: This PR concerns both.
The issue with import ordering only exists for `generate` I think.

**Comment by carbonin** [comment]
closing this. Will re-open and rebase if it becomes an issue again

**Pull request closed without merging**